### ISSUE TITLE
[rustdoc] Common well-known traits make "Trait Implementations" too verbose

### ISSUE THREAD

**Issue by pornel** [rust-lang/rust#40363]
I find the "Trait Implementations" section of Rustdoc very hard to follow. It has low signal to noise ratio:

* Very common traits are displayed in full detail. For traits like `Copy`, `Clone`, `Default`, `Display`, `Debug`, `Hash`, etc. I don't need to see all the methods *inline* with their description. Just links to these traits would be enough. Generally I only need to know whether the type implements them or not.

* Similarly, traits used via operators are also displayed in full detail. `Eq`, `PartialEq`, `Ord`, `PartialOrd`, `Add`, `Sub`, etc. flood the documentation page with documentation *of* the traits. The long verbose list makes it hard to see *which* traits are supported.

* If the type implements the same trait multiple times for slightly different types, the whole trait documentation is repeated multiple times. It is useful to know which type combinations are supported, but repeated function descriptions only lower signal to noise ratio.

The current layout does work well for less common/more specific traits, e.g. having entire trait documentation inlined is perfect for `*Ext` traits and makes sense in some cases like `impl Read for File`.

So I'm suggesting:

* Implement collapsing widget for trait implementations, which collapses documentation of methods, leaving only `[+] impl Trait for Foo` line.

* Collapse basic traits by default (`Copy`, `Clone`, `Debug`, `std::ops::*`, etc.). Not all traits in `std` are obvious, so the list of traits to display in a terse form would probably be manually curated.

* Group implementations of the same trait together, and collapse all implementations except one. For example [`File`](https://doc.rust-lang.org/std/fs/struct.File.html) displays all of `Read` docs twice, in two separate places. It makes the list longer and makes it harder to notice that there are two implementations. It would be clearer to display it as:

    > `[+] impl Read for File`
    > `[-] impl<'a> Read for &'a File`
    >  ` fn read(&mut self) …`

**Comment by GuillaumeGomez** [comment]
cc @rust-lang/docs

**Comment by vi** [comment]
> Implement collapsing widget for trait implementations, which collapses documentation of methods, leaving only `[+] impl Trait for Foo` line.

Thinking about implementing this.

**Comment by vi** [comment]
Is there some exhausive list or traits (in `std` or maybe across the entire ecosystem) that should be collapsed bydefault?

**Comment by kornelski** [comment]
I'd look through `std::ops`, as these are the noisiest.

**Comment by vi** [comment]
It is also suggested just to collapse all impls items by default.
